core/src/impl/Kokkos_Atomic_Generic.hpp

```diff
@@ -78,6 +86,10 @@ struct AddOper {
   static Scalar1 apply(const Scalar1& val1, const Scalar2& val2) {
     return val1 + val2;
   }
+  KOKKOS_FORCEINLINE_FUNCTION
+  static bool test(const Scalar1& val1, const Scalar2& val2) {
+    return false;
```

Comment:
We test with -Werror and this static member functions and all the ones below trigger  unused parameter warnings.

Reply:
As you can see from how it is used, it has to take the arguments.  If it does anything with them, it means adding unnecessary instructions.

I assume I cannot use C++17 `[[maybe_unused]]`.  Does Kokkos have a macro that behaves that way?

Reply:
> As you can see from how it is used, it has to take the arguments. If it does anything with them, it means adding unnecessary instructions.

You can omit the name of the parameters (preferred here) or cast them to void in the body of the function.

> I assume I cannot use C++17 `[[maybe_unused]]`.

No unfortunately we still support C++11. 

>  Does Kokkos have a macro that behaves that way?

We do not.

Reply:
I implemented the suggestion of @masterleinad